### PR TITLE
fix: Use better layouts to better support RTL in messages

### PR DESCRIPTION
**Pull request by JPonte**
fixes: https://github.com/wireapp/android-project/issues/328
#### APK
[Download build #12083](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12083/artifact/build/artifact/wire-dev-PR1871-12083.apk)
[Download build #12092](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12092/artifact/build/artifact/wire-dev-PR1871-12092.apk)
[Download build #12093](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12093/artifact/build/artifact/wire-dev-PR1871-12093.apk)
[Download build #12104](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12104/artifact/build/artifact/wire-dev-PR1871-12104.apk)
[Download build #12120](http://192.168.120.33:8080/job/Pull%20Request%20Builder/12120/artifact/build/artifact/wire-dev-PR1871-12120.apk)

### CODE REVIEW

File: app/src/main/res/layout/message_separator_large.xml [hunk 1]
```diff
@@ -27,21 +27,25 @@
 
     <com.waz.zclient.messages.UnreadDot
         android:id="@+id/unread_dot"
-        android:layout_width="@dimen/content__padding_left"
+        android:layout_width="72dp"
```
Comment (deanrobertcook) [inline app/src/main/res/layout/message_separator_large.xml]: Should we perhaps be using resource values?

File: app/src/main/res/layout/system_message_content.xml [hunk 2]
```diff
@@ -27,36 +27,40 @@
 
     <com.waz.zclient.ui.text.GlyphTextView
         android:id="@+id/gtv__system_message__icon"
-        android:layout_marginStart="@dimen/system_margin_start"
-        android:layout_marginEnd="@dimen/system_margin_end"
         android:layout_width="@dimen/wire__otr__shield__small_width"
         android:layout_height="@dimen/wire__otr__shield__small_width"
-        android:gravity="center"
         android:layout_gravity="center"
+        android:layout_marginStart="16dp"
```
Comment (deanrobertcook) [inline app/src/main/res/layout/system_message_content.xml]: and here

File: app/src/main/res/layout/system_message_content.xml [hunk 3]
```diff
@@ -27,36 +27,40 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"/>
 
     <com.waz.zclient.common.views.LinkTextView
         android:id="@+id/ttv__system_message__text"
         android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_toEndOf="@id/gtv__system_message__icon"
+        android:layout_height="18dp"
+        android:layout_marginStart="32dp"
```
Comment (deanrobertcook) [inline app/src/main/res/layout/system_message_content.xml]: and here

**Comment by zeljkotutic** [comment]
build again

**Comment by zeljkotutic** [comment]
There are still few issues in RTL layout:

screen1

- name of user whole liked message is on the wrong side and not near like (heart) icon (like on screenshot)
- quoted message layout is wrong also (like on screenshot)
- message input text box cursor is firstly on the right (before you start writting) and then whole input goes from the left (like on screenshot)
- ephemeral counter icon is on the left (like on usual left to right layout) and not on the right side near the message

**Pull request closed without merging**